astropy/table/operations.py

```diff
@@ -87,9 +88,238 @@ def _get_out_class(objs):
     return out_class
 
 
+def join_skycoord(distance, distance_kind='sky'):
+    """Helper function to join on SkyCoord columns using distance matching.
+
+    This function is intended for use in ``table.join()`` to allow performing a
+    table join where the key columns are both ``SkyCoord`` objects, matched by
+    computing the distance between points and accepting values below
+    ``distance``.
+
+    The distance cross-matching is done using either
+    `~astropy.coordinates.SkyCoord.search_around_sky` or
+    `~astropy.coordinates.SkyCoord.search_around_3d`, depending on the value of
+    ``distance_kind``.
+
+    Parameters
+    ----------
+    distance : Quantity (angle or length)
+        Maximum distance between points to be considered a join match
+    distance_kind : str
```

Comment:
In principle, this could be inferred based on the unit of `distance`.

Reply:
Good idea, will do.

Reply:
Even then I much prefer one less kwarg (*simple is better than complex*), on further reflection I think this "guessing" may be ambiguous because of `u.dimensionless_angles()`. Because of this equivalency one can pass in a dimensionless Quantity and it could either be a 3d cartesian distance or a dimensionless angular distance.  I don't know of any rock solid way to disambiguate, and even if we come up with one it will be complicated to explain to users. So I think it might be best to just leave this be (*explicit is better than implicit*).

Reply:
Can dimensionless ever be a 3-d distance? I guess the test would be `...unit.is_equivalent(u.radian)`. 

But perhaps an alternative is to have the default be determined by the unit? It is good to have the user be able to pass in an explicit `distance_kind` so that things will simply fail if `distance` has the wrong units.